### PR TITLE
update kind to 0.20.0

### PR DESCRIPTION
**Pull request by kannon92**
#### What type of PR is this?
/kind cleanup
/kind testing

#### What this PR does / why we need it:
Update Kind to 0.20.0 so we can pick up 1.29.0 images.

Drop deprecated kubernetes versions.
#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:
I am noticing something odd in Kind 0.20.0 around cert rotation so I think we should test the CI pretty well here once 1.29 test-infra PR merges.

See https://github.com/kubernetes-sigs/jobset/issues/357 for some discussion on this.
#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -65,7 +65,7 @@ E2E_TARGET ?= ./test/e2e/...
 E2E_KIND_VERSION ?= kindest/node:v1.28.0
 
 # E2E_K8S_VERSIONS sets the list of k8s versions included in test-e2e-all
-E2E_K8S_VERSIONS ?= 1.24.15 1.25.11 1.26.6 1.27.3 1.28.0
+E2E_K8S_VERSIONS ?= 1.26.6 1.27.3 1.28.0 1.29.0
```
Comment (tenzen-y) [inline Makefile:68]: @kannon92 Can we update the Kubernetes version in other places like `ENVTEST_K8S_VERSION`?

**Comment by kannon92** [comment]
@tenzen-y @alculquicondor I got a bit confused by kind's website. I was thinking that the latest versions of Kind would have 1.29 but it seems that 0.18.0 has it also.

Do you all still an update to the latest version of kind?

File: Makefile (hunk 1 above)
Comment (kannon92) [reply Makefile:68]: Its at 1.28.0 and I don't know if we should update that to 1.29.0 yet.

**Comment by tenzen-y** [comment]
> @tenzen-y @alculquicondor I got a bit confused by kind's website. I was thinking that the latest versions of Kind would have 1.29 but it seems that 0.18.0 has it also.
> 
> Do you all still an update to the latest version of kind?

@kannon92 I see. Then, can we update only the KinD version in this PR?

File: Makefile (hunk 1 above)
Comment (alculquicondor) [reply Makefile:68]: let's use the latest patch version for 1.28

Comment (kannon92) [reply Makefile]: updated. sorry for the lag.

**Comment by kannon92** [comment]
> > @tenzen-y @alculquicondor I got a bit confused by kind's website. I was thinking that the latest versions of Kind would have 1.29 but it seems that 0.18.0 has it also.
> > Do you all still an update to the latest version of kind?
> 
> @kannon92 I see. Then, can we update only the KinD version in this PR?

@alculquicondor and @tenzen-y I pushed a change for this and updated all the kind versions to be the latest kind and updated the version. It probably makes sense for kind image to be the latest patch.

**Review by tenzen-y — COMMENTED** [review]
Thanks!
/lgtm
/apprvoe

**Pull request merged**